ql/src/test/queries/clientpositive/sketches_materialized_view_cume_dist.q

```diff
@@ -0,0 +1,54 @@
+--! qt:transactional
+set hive.fetch.task.conversion=none;
+
+create table sketch_input (id int, category char(1))
+STORED AS ORC
+TBLPROPERTIES ('transactional'='true');
+
+insert into table sketch_input values
+  (1,'a'),(1, 'a'), (2, 'a'), (3, 'a'), (4, 'a'), (5, 'a'), (6, 'a'), (7, 'a'), (8, 'a'), (9, 'a'), (10, 'a'),
+  (6,'b'),(6, 'b'), (7, 'b'), (8, 'b'), (9, 'b'), (10, 'b'), (11, 'b'), (12, 'b'), (13, 'b'), (14, 'b'), (15, 'b')
+; 
+
+-- create an mv for the intermediate results
+create  materialized view mv_1 as
+  select category,ds_kll_sketch(cast(-id as float)) from sketch_input group by category;
+
+-- bi mode on
+set hive.optimize.bi.enabled=true;
+
+explain
+select 'rewrite; mv matching', id, cume_dist() over (order by id) from sketch_input order by id;
+select 'rewrite; mv matching', id, cume_dist() over (order by id) from sketch_input order by id;
```

Comment:
it would be handy to introduce an instruction that the next sql statement "must" touch the table "mv_1"

```
--! qt:plan:mustreadtable:mv_1
```
I don't know implementing something like this is how much of a science fiction - it's certainly possible; however wiring that in might be tricky...

Reply:
Yes, that's certainly useful. Something that may help for explain plans is to use `explain cbo` as it is definitely less verbose.